internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -148,14 +148,20 @@ func restartAgent(ctx context.Context) error {
 
 	for i := maxRestartCount; i >= 1; i-- {
 		backExp.Wait()
+		attempt := maxRestartCount - i + 1
```

Comment:
wouldnt it be easier and more optimal to switch for logic and condition on line 159?

Reply:
You mean to make the loop go from 0 to `maxRestartCount` instead of how it is now, going in the opposite direction?

Reply:
@michalpristas I made a change to the logic in 5ec79d4ca. Is this what you were suggesting?

Reply:
yeah, much nicer and no arithmetic operations on top of the loop